pyresample/geometry.py

```diff
@@ -146,25 +146,30 @@ def __eq__(self, other):
             self_lons = self.lons
             self_lats = self.lats
 
-        if self_lons is other_lons and self_lats is other_lats:
-            return True
         if isinstance(self_lons, DataArray) and np.ndarray is not DataArray:
             self_lons = self_lons.data
             self_lats = self_lats.data
         if isinstance(other_lons, DataArray) and np.ndarray is not DataArray:
             other_lons = other_lons.data
             other_lats = other_lats.data
+        if self_lons is other_lons and self_lats is other_lats:
+            return True
```

Comment:
Why was this block moved from before to after the `isinstance` checks?

Reply:
Ah because now it checks if the dask arrays are the same instance. Before it would only check if the DataArrays were the same. This is more likely to catch more cases (ex. two DataArrays with the same dask array underneath).